src/dex/uniswap-v2/tetuswap.ts

```diff
@@ -0,0 +1,120 @@
+import { UniswapV2, UniswapV2Pair, UniswapV2PoolState } from './uniswap-v2';
+import { Network } from '../../constants';
+import { Address, DexConfigMap } from '../../types';
+import { IDexHelper } from '../../dex-helper';
+import { AbiCoder, Interface } from '@ethersproject/abi';
+import { DexParams } from './types';
+import { getDexKeysWithNetwork } from '../../utils';
+import _ from 'lodash';
+import TetuSwapPoolABI from '../../abi/uniswap-v2/tetuswap-pool.json';
+
+const coder = new AbiCoder();
+
+export const TetuSwapConfig: DexConfigMap<DexParams> = {
+  TetuSwap: {
+    [Network.POLYGON]: {
+      subgraphURL: 'https://api.thegraph.com/subgraphs/name/tetu-io/tetu-swap',
+      factoryAddress: '0x684d8c187be836171a1Af8D533e4724893031828',
+      router: '0xBCA055F25c3670fE0b1463e8d470585Fe15Ca819',
+      // To PR reviewer:
+      // TetuSwap init code is different for different pairs
+      // does it need additional configuration to use buyOnUniswapV2Fork only
+      // and do not use buyOnUniswapFork method that needs initCode?
+      initCode: '0x0',
+      poolGasCost: 1000 * 1000, // TetuSwap use SmartVault deposits / withdrawals during swap, so its costly
```

Comment:
Is it really `1 million` cost?

Reply:
it different, but about 1million  - because it invest / withdraws from yeld farming strategies under the hood. See router's swap transactions https://polygonscan.com/address/0xbca055f25c3670fe0b1463e8d470585fe15ca819